code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -287,13 +289,18 @@
 	update_button_icon()
 
 /datum/action/xeno_action/activable/secrete_resin/use_ability(atom/A)
+	//xeno owner of this ability
 	var/mob/living/carbon/xenomorph/xowner = owner
-	if(SSmonitor.gamestate == SHUTTERS_CLOSED && CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD) && SSresinshaping.active)
+	var/turf/selectedturf = get_turf(xowner)
+	if(SSmonitor.gamestate != SHUTTERS_CLOSED)
+		if(selectedturf.get_build_prereqs())
```

Comment:
why are we checking get_build_prereqs if shutters have opened?

Reply:
I was tired when I made this, hopefully the rewrite is closer to what you intended.